src/runtime/cpu_device_api.cc

```diff
@@ -32,8 +35,20 @@ class CPUDeviceAPI final : public DeviceAPI {
     ptr = memalign(alignment, nbytes);
     if (ptr == nullptr) throw std::bad_alloc();
 #else
-    int ret = posix_memalign(&ptr, alignment, nbytes);
-    if (ret != 0) throw std::bad_alloc();
+#ifndef _WIN32
```

Comment:
Can we add a compilation flag to control whether to turn hugepage on or off.

Reply:
why do you want to disable hugepage?

Reply:
I don't think hugepage is available on Mac.  It does seem better to me to have a compilation flag instead.

Reply:
then we should only enable hugepage in linux.